docs/selection_example.py

```diff
@@ -0,0 +1,203 @@
+                mutation_type_id=mut_id, population_id=pop["Papuan"],
+                op=">", allele_frequency=0,
+                # Update save point at start_time (if the condition is met).
+                # If the Den1 migrants didn't carry the mutation, we will
+                # instead restore to the previous save point.
+                save=True),
+        # The mutation is positively selected in Papuans at T_sel.
+        # Note that this will have no effect, unless/until a mutation with the
+        # specified mutation_type_id is found in the population.
+        stdpopsim.ext.ChangeMutationFitness(
+                start_time=T_sel, end_time=0,
+                mutation_type_id=mut_id, population_id=pop["Papuan"],
+                selection_coeff=s, dominance_coeff=0.5),
+        # Condition on AF > 0.05 in Papuans at the end of the simulation.
+        stdpopsim.ext.ConditionOnAlleleFrequency(
+                start_time=0, end_time=0,
+                mutation_type_id=mut_id, population_id=pop["Papuan"],
+                op=">", allele_frequency=0.05),
+        ]
+
+    # Simulate.
+    engine = stdpopsim.get_engine("slim")
+    ts = engine.simulate(
+            model, contig, samples,
+            seed=rng.randrange(1, 2**32),
+            mutation_types=mutation_types,
+            extended_events=extended_events,
+            slim_scaling_factor=10,
+            slim_burn_in=0.1,
+            # Set slim_script=True to print the script instead of running it.
+            # slim_script=True,
+            )
+
+    return ts, T_mut, T_sel, s
+
+
+def KimDFE():
+    """
+    Return neutral and negative MutationType()s representing a human DFE.
+    Kim et al. (2018), p.23, http://doi.org/10.1371/journal.pgen.1007741
+    """
+    neutral = stdpopsim.ext.MutationType(weight=1.0)
+    gamma_shape = 0.186  # shape
+    gamma_mean = -0.01314833  # expected value
+    h = 0.5/(1-7071.07*gamma_mean)  # dominance coefficient
+    negative = stdpopsim.ext.MutationType(
+            weight=2.31, dominance_coeff=h, distribution_type="g",
+            distribution_args=[gamma_mean, gamma_shape])
+    return [neutral, negative]
+
+
+def OutOfAfrica_3G09_with_DFE(seed):
+    """
+    The Gutenkunst et al. HomSap/OutOfAfrica_3G09 model, simulated with a DFE.
+    """
+    species = stdpopsim.get_species("HomSap")
+    model = species.get_demographic_model("OutOfAfrica_3G09")
+    contig = species.get_contig("chr1", length_multiplier=0.001)
+    samples = model.get_samples(
+            100,  # YRI
+            100,  # CEU
+            100)  # CHB
+
+    mutation_types = KimDFE()
+
+    # Simulate.
+    engine = stdpopsim.get_engine("slim")
+    ts = engine.simulate(
+            model, contig, samples,
+            seed=seed,
+            mutation_types=mutation_types,
+            slim_scaling_factor=10,
+            slim_burn_in=10,
+            # Set slim_script=True to print the script instead of running it.
+            # slim_script=True,
```

Comment:
@petrelharp: pull down the pr branch, then edit this line of `docs/selection_example.py`, and run it to emit the slim script to stdout.